cmd/clusterctl/pkg/internal/test/fake_objects.go

```diff
@@ -775,15 +877,57 @@ func (f *FakeMachine) Objs(cluster *clusterv1.Cluster, generateCerts bool, machi
 	return objs
 }
 
-func reconcileMachineExternalRef(externalRef *unstructured.Unstructured, machine *clusterv1.Machine) {
-	externalRef.SetOwnerReferences([]metav1.OwnerReference{
-		{
-			APIVersion: machine.APIVersion,
-			Kind:       machine.Kind,
-			Name:       machine.Name,
+// setUID assigns a UID to the object, so test objects are uniquely identified.
+// NB. In order to make debugging easier we are using a human readable, deterministic string (instead of a random UID).
+func setUID(obj runtime.Object) {
+	accessor, _ := meta.Accessor(obj)
+	uid := fmt.Sprintf("%s, %s/%s", obj.GetObjectKind().GroupVersionKind().String(), accessor.GetNamespace(), accessor.GetName())
+	accessor.SetUID(types.UID(uid))
+}
+
+// FakeCustomResourceDefinition returns a fake CRD object for the given group/versions/kind.
+func FakeCustomResourceDefinition(group string, kind string, versions ...string) *apiextensionslv1.CustomResourceDefinition {
+	crd := &apiextensionslv1.CustomResourceDefinition{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       apiextensionslv1.SchemeGroupVersion.String(),
+			APIVersion: "CustomResourceDefinition",
 		},
-	})
-	externalRef.SetLabels(map[string]string{
-		clusterv1.ClusterLabelName: machine.Spec.ClusterName,
-	})
+		ObjectMeta: metav1.ObjectMeta{
+			Name: fmt.Sprintf("%s.%s", strings.ToLower(kind), group),
```

Comment:
Technically the name needs to be pluralized if you want to match real CRD names. Do you want to do that?

Reply:
For the sake of test what really matters is to generate a unique name, so I think we can leave as it is